### PR TITLE
JBTM-2583 Adding JTAActionStatusServiceXAResourceOrphanFilter

### PR DESCRIPTION
**Pull request by primIl**
Adding JTAActionStatusServiceXAResourceOrphanFilter to configured orphan filters

### CODE REVIEW

File: atsintegration/examples/classes/com/arjuna/jta/distributed/example/server/impl/ServerImpl.java [hunk 1]
```diff
@@ -151,6 +151,7 @@ public void initialise(LookupProvider lookupProvider, String nodeName, int portO
 		List<String> xaResourceOrphanFilterClassNames = new ArrayList<String>();
 
 		xaResourceOrphanFilterClassNames.add("com.arjuna.ats.internal.jta.recovery.arjunacore.JTATransactionLogXAResourceOrphanFilter");
+		xaResourceOrphanFilterClassNames.add("com.arjuna.ats.internal.jta.recovery.arjunacore.JTAActionStatusServiceXAResourceOrphanFilter");
 		xaResourceOrphanFilterClassNames.add("com.arjuna.ats.internal.jta.recovery.arjunacore.JTANodeNameXAResourceOrphanFilter");
 		xaResourceOrphanFilterClassNames.add("com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateJTAXAResourceOrphanFilter");
```
Comment (ochaloup) [inline atsintegration/examples/classes/com/arjuna/jta/distributed/example/server/impl/ServerImpl.java]: for the sake of the completeness, can you please add down here the `JTAActionStatus...` filter for subordinate transactions too? It's class `com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinationManagerXAResourceOrphanFilter`

**Comment by ochaloup** [comment]
@mmusgrov Mike I would like to check you one nuance. Here the order of adding orphan filters is

```
1. JTATransactionLogXAResourceOrphanFilter
2. JTAActionStatusServiceXAResourceOrphanFilter
3. JTANodeNameXAResourceOrphanFilter`
```

I can see the order in the WFLY is

```
1. JTATransactionLogXAResourceOrphanFilter
2. JTANodeNameXAResourceOrphanFilter
3. JTAActionStatusServiceXAResourceOrphanFilter
```

as checking the code I think the order is not in importance here, but would you be so kind and confirm if that is ok or should be the same as in WFLY? Thanks in advance!

**Comment by ochaloup** [comment]
@primIl thank you! I would just like to wait for @mmusgrov to confirm my doubt about ordering before this being merge.

**Comment by mmusgrov** [comment]
@ochaloup The order of the orphan filters should not matter since we check them all and only then make a decision on what to do based on the interface guarantee provided by XAResourceOrphanFilter. However, from a defensive programming point of view it could be that one filter has side effects that affect the behaviour of another filter (that would be bad programming practice but feasible) so personally I would strive to keep the order consistent.

**Comment by ochaloup** [comment]
Issue is caused by changes in WFLY which was fixed by CI sh script change - https://github.com/jbosstm/narayana/pull/1282. It should be working now.

**Comment by ochaloup** [comment]
@primIl would you be so kind and squash your 3 commits under one, please?

**Comment by ochaloup** [comment]
@primIl I found out that the same script fix is needed not just in the Narayana but in the interop quickstart. I pushed my change here, waiting to be retested.

**Comment by ochaloup** [comment]
@primIl ok, all works fine now. the last think, please squash your changes under one commit. Thanks in advance!

**Comment by ochaloup** [comment]
@primIl perfect. many thanks for the fix!

**Pull request merged**